atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampStoreInvalidatorIntegrationTest.java

```diff
@@ -0,0 +1,155 @@
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.cassandra.CassandraKeyValueServiceConfigManager;
+import com.palantir.atlasdb.containers.CassandraContainer;
+import com.palantir.atlasdb.containers.Containers;
+import com.palantir.common.base.Throwables;
+import com.palantir.timestamp.TimestampBoundStore;
+import com.palantir.timestamp.TimestampStoreInvalidator;
+
+public class CassandraTimestampStoreInvalidatorIntegrationTest {
+    @ClassRule
+    public static final Containers CONTAINERS = new Containers(CassandraTimestampStoreInvalidatorIntegrationTest.class)
+            .with(new CassandraContainer());
+
+    private static final long ZERO = 0L;
+    private static final long ONE_MILLION = 1000000L;
+    private static final int POOL_SIZE = 16;
+    private static final int TIMEOUT_SECONDS = 5;
+
+    private final CassandraKeyValueService kv = CassandraKeyValueService.create(
+            CassandraKeyValueServiceConfigManager.createSimpleManager(CassandraContainer.KVS_CONFIG),
+            CassandraContainer.LEADER_CONFIG);
+    private final TimestampStoreInvalidator invalidator = new CassandraTimestampStoreInvalidator(kv);
+
+    private TimestampBoundStore timestampBoundStore;
+    private ExecutorService executorService;
+
+    @Before
+    public void setUp() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+        timestampBoundStore = CassandraTimestampBoundStore.create(kv);
+        executorService = Executors.newFixedThreadPool(POOL_SIZE);
+    }
+
+    @After
+    public void close() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+        kv.close();
+    }
+
+    @Test
+    public void canInvalidateTimestampTableIfItAlreadyExistsWithData() {
+        takeOutOneMillionTimestamps();
+        invalidateAndThenCheckCannotReadBound();
+    }
+
+    @Test
+    public void canInvalidateTimestampTableIfItAlreadyExistsWithoutData() {
+        invalidateAndThenCheckCannotReadBound();
+    }
+
+    @Test
+    public void canInvalidateTimestampTableIfItDoesNotExistYet() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+        invalidateAndThenCheckCannotReadBound();
+    }
+
+    @Test
+    public void invalidateIsIdempotent() {
+        invalidator.invalidateTimestampStore();
+        invalidateAndThenCheckCannotReadBound();
+    }
+
```

Comment:
Sure, that improves the coverage! I think I initially thought of this as "invalidate 2x is safe", but you're right that this is a better way to check that it's really idempotent.